### PR TITLE
NCSDK-24382: Add support for SHA512

### PR DESCRIPTION
**Pull request by tomchy**
Add support for SHA512 as component and manifest digest.

### CODE REVIEW

**Review by robertstypa — APPROVED** [review]

File: src/suit.c [hunk 1]
```diff
@@ -316,14 +316,32 @@ int suit_processor_get_manifest_metadata(uint8_t *envelope_str, size_t envelope_
 		ret = cbor_decode_SUIT_Digest(digest_bstr->value, digest_bstr->len, &digest_cbor,
 					      &bytes_processed);
 
-		/* The SHA256 algorithm is enforced by CDDL, so the digest length is known. */
-		if ((bytes_processed != digest_bstr->len) ||
-		    (digest_cbor._SUIT_Digest_suit_digest_bytes.len != 32)) {
+		if (bytes_processed != digest_bstr->len) {
 			ret = SUIT_ERR_DECODING;
```
Comment (rafp83) [inline src/suit.c:320]: Considering in all cases returned error is the same, why only store the error code and not return?

File: include/suit_legacy.h
Comment (adsz-nordic) [inline include/suit_legacy.h:1]: Is this file kept empty for some reason or should it be deleted?

File: src/suit.c (hunk 1 above)
Comment (tomchy) [reply src/suit.c:320]: To have a common clean-up code:
```
decoder_state->step = INVALID;
```

**Review by adsz-nordic — COMMENTED** [review]
Should include/suit_legacy.h be kept empty?

File: include/suit_legacy.h
Comment (tomchy) [reply include/suit_legacy.h:1]: It is empty and it is deleted by this PR

Comment (adsz-nordic) [reply include/suit_legacy.h:1]: OK. I asked since normally github marks deleted files explicitly and this one is not marked in such a way.

**Review by adsz-nordic — APPROVED** [review]

File: src/suit.c (hunk 1 above)
Comment (ahasztag) [reply src/suit.c:320]: Then shouldn't all the following if clauses do the same?

Comment (tomchy) [reply src/suit.c:320]: Ah, now I understand - all other branches aligned! :+1:

**Review by ahasztag — APPROVED** [review]

**Pull request merged**